session/session.go

```diff
@@ -2372,6 +2373,17 @@ func runStmt(ctx context.Context, se *session, s sqlexec.Statement) (rs sqlexec.
 			se:        se,
 		}, err
 	}
+	if !se.GetSessionVars().InRestrictedSQL && se.GetSessionVars().IsPlanReplayerCaptureEnabled() {
+		stmtNode := s.GetStmtNode()
+		startTS := se.GetSessionVars().StmtCtx.StmtSnapshotTS
+		if se.GetSessionVars().EnablePlanReplayedContinuesCapture {
+			if checkPlanReplayerContinuesCaptureValidStmt(stmtNode) {
+				checkPlanReplayerContinuesCapture(se, stmtNode, startTS)
+			}
+		} else {
+			checkPlanReplayerCaptureTask(se, stmtNode, startTS)
+		}
+	}
```

Comment:
Can we add this logic into `ExecStmt.FinishExecuteStmt`?

Reply:
updated.